### PR TITLE
Test config: don't allow HOMEBREW_BREW_FILE to be unset

### PR DESCRIPTION
**Pull request by mistydemeo**

### CODE REVIEW

File: Library/Homebrew/test/lib/config.rb [hunk 1]
```diff
@@ -1,7 +1,12 @@
 require "tmpdir"
 require "pathname"
 
-HOMEBREW_BREW_FILE = Pathname.new(ENV["HOMEBREW_BREW_FILE"])
+# HOMEBREW_BREW_FILE is most likely unset if this is called via `rake test`
+begin
+  HOMEBREW_BREW_FILE = Pathname.new(ENV.fetch("HOMEBREW_BREW_FILE"))
+rescue KeyError
+  HOMEBREW_BREW_FILE = Pathname.new(File.expand_path("../../../../../bin/brew", __FILE__))
+end
```
Comment (xu-cheng) [inline Library/Homebrew/test/lib/config.rb]: You can drop the rescue block like this:
```ruby
HOMEBREW_BREW_FILE = Pathname.new \
  ENV.fetch("HOMEBREW_BREW_FILE", File.expand_path("../../../../../bin/brew", __FILE__))
```

**Comment by UniqMartin** [comment]
Personally, I would prefer if we just died/aborted in case the environment variable is not set. Trying to automatically recover has the potential to create more confusion/damage, as it will mask other errors where that environment variable is not set. (After all this isn't in user-facing code, so it's better to fail that to apply magic.)

**Comment by mistydemeo** [comment]
Updated to just die if it's unset, mimicing the message from `config.rb`.

**Comment by UniqMartin** [comment]
Is there any benefit in relying on exceptions here instead of using explicit control flow [like in `config.rb`](https://github.com/Homebrew/homebrew/blob/c3438ae77441c606e75b1c3697589588177f75ad/Library/Homebrew/config.rb#L29-L33)? I think keeping them aligned makes the code just a tiny bit less confusing. Otherwise :+1:.

**Comment by apjanke** [comment]
I think this is good to go, maybe with Martin's suggestion. Do you mind closing and re-submitting this over at homebrew/brew?

**Comment by MikeMcQuaid** [comment]
Can this be ported over to the Homebrew/brew repo when it's ready to 🚢? Thanks!

**Pull request closed without merging**